actix_todo/src/api.rs

```diff
@@ -0,0 +1,174 @@
+use actix::prelude::Addr;
+use actix_web::middleware::Response;
+use actix_web::{
+    error, fs::NamedFile, http, AsyncResponder, Form, FutureResponse, HttpRequest,
+    HttpResponse, Path, Responder, Result,
+};
+use futures::{future, Future};
+use tera::{Context, Tera};
+
+use db::{AllTasks, CreateTask, DbExecutor, DeleteTask, ToggleTask};
+use session::{self, FlashMessage};
+
+pub struct AppState {
+    pub template: Tera,
+    pub db: Addr<DbExecutor>,
+}
+
+pub fn index(req: HttpRequest<AppState>) -> FutureResponse<HttpResponse> {
+    req.state()
+        .db
+        .send(AllTasks)
+        .from_err()
+        .and_then(move |res| match res {
+            Ok(tasks) => {
+                let mut context = Context::new();
+                context.add("tasks", &tasks);
+
+                if let Some(flash) = session::get_flash(&req)? {
```

Comment:
Minor thing: could you please add here brief comment that flash is used to propagate data on redirect from other endpoints